### PR TITLE
Feature/1240 map test dat from json file to java object

### PR DESCRIPTION
**Pull request by MamdouhMansour**

### CODE REVIEW

**Comment by MamdouhMansour** [comment]
Related to #1240

File: src/main/java/objectclass/BodyObject.java
Comment (MohabMohie) [inline src/main/java/objectclass/BodyObject.java:1]: Question...
Is it possible to build this class dynamically in run time after parsing the json file?
@MhmdElGazzar supervised a similar implementation with the CuRL to POJO generator, is it possible to create a dynamic json to POJO parser? better-yet, did someone already do this and can we just use it as a dependency?

To me, it makes little sense having a static mapping with static json property names for test data... we have a similar static implementation for SHAFT properties, but for test data I think we need a dynamic implementation.


aside from that, this looks like an excellent implementation. Thanks Mamdouh.

**Review by MohabMohie — CHANGES_REQUESTED** [review]
one minor change request, can we move the BodyObject.java to the src/test/java directory since it's not a part of the engine (because it has static mapping of the specific test data we're using to test this feature).

File: src/main/java/objectclass/BodyObject.java
Comment (MamdouhMansour) [reply src/main/java/objectclass/BodyObject.java:1]: @MhmdElGazzar @MohabMohie would like to know more to implement

**Comment by MamdouhMansour** [comment]
@MohabMohie moved pojo out of the engine

**Comment by MamdouhMansour** [comment]
@MohabMohie what we need to complete this PR?

**Review by MohabMohie — APPROVED** [review]

**Pull request merged**